new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/BuildApplication.java

```diff
@@ -174,8 +175,14 @@ public static Path buildApplication(Path appSrcPath, Map<String, String> antPara
       if (exec.stderr() != null) {
         logger.info("Exec stderr {0}", exec.stderr());
       }
-      assertEquals(0, exec.exitValue(), "Exec into " + webLogicPod.getMetadata().getName()
-          + " to build an application failed with exit value " + exec.exitValue());
+
+      // Exec returns a non-zero return code intermittently even when the application builds
+      // successfully. This seems to be an issue with io.kubernetes.client.Exec.java. So, Commenting
+      // this assertion for now. it is now the responsibility of the It test class
+      // to assert the application exists before continuing with the test.
```

Comment:
why can't the check be done in this method instead of changing all the calling It test classes?

Reply:
The source application might build more than 1 archive and we don't know what file name it will be, so it is better to leave it to the It class to do that.

Reply:
I see the stderr is empty when it returns non-zero code on success, how about checking for 0 exit value or (non-zero and stderr is empty) for success? assuming on failure, stderr is not empty, this needs to be checked if not sure.
https://build.weblogick8s.org:8443/job/wko-kind-nightly/95/artifact/logdir/jenkins-wko-kind-nightly-95/wl_k8s_test_results/diagnosticlogs/ItCrossDomainTransaction/ItCrossDomainTransaction.out/*view*/

Reply:
I don't know if we can rely on that logic. IMO a simple assertion in the It class would be sufficient. Like this
`assertTrue(clusterViewAppPath.toFile().exists(), "Application archive is not available");`

Reply:
Is that check done in all the existing it classes which calls this method? If not, that needs to be done in the It classes as part of this PR as this assertion is removed here, otherwise tests will proceed even if the application build fails.

Reply:
Only 1 class was missing the assert. Added the assertion in ItManagedCoherence.java